client/signup/steps/plans/index.jsx

```diff
@@ -46,16 +47,24 @@ class PlansStep extends Component {
 			} );
 		}
 
-		SignupActions.submitSignupStep( {
+		const step = {
 			processingMessage: isEmpty( cartItem )
 				? translate( 'Free plan selected' )
 				: translate( 'Adding your plan' ),
 			stepName,
 			stepSectionName,
 			cartItem,
-			privacyItem,
 			...additionalStepData
-		}, [], { cartItem, privacyItem } );
+		};
+
+		const providedDependencies = { cartItem };
+
+		if ( privacyItem ) {
```

Comment:
I'm not very familiar with the signup framework, so I'm sorry if that was not the proper way to handle that.

The reason that I made it here is because we doing the following:

In [onSelectPlan](https://github.com/Automattic/wp-calypso/pull/11224/files#diff-226cf00033910ff5985fa4e54353707aR67) we call SignupActions.[submitSignupStep](https://github.com/Automattic/wp-calypso/blob/master/client/lib/signup/actions.js#L28) that dispatches the `SUBMIT_SIGNUP_STEP ` action, that is handled in [SignupDependencyStore](https://github.com/Automattic/wp-calypso/blob/master/client/lib/signup/dependency-store.js#L58) which performs an [assertion](https://github.com/Automattic/wp-calypso/blob/master/client/lib/signup/dependency-store.js#L38) that no excessive dependencies are provided by a step. 

 Which results in:
![screen shot 2017-02-09 at 10 25 18](https://cloud.githubusercontent.com/assets/326402/22775020/6fdce918-eeb2-11e6-857a-13f5fde9742e.png)

Again, I'm not really familiar with the signup, but later we [dispatch](https://github.com/Automattic/wp-calypso/blob/master/client/lib/signup/dependency-store.js#L60) `SIGNUP_DEPENDENCY_STORE_UPDATE` to redux, that overwrites everything that is 'provided' with the new data, so if we have something extra that is falsy / wrong, we might be [overwriting](https://github.com/Automattic/wp-calypso/blob/master/client/state/signup/dependency-store/reducer.js#L15) something that we shouldn't, so that assertion looks ok to me.